yarn-project/types/src/l1_to_l2_message.ts

```diff
@@ -17,6 +17,79 @@ export interface L1ToL2MessageSource {
   getPendingL1ToL2Messages(take: number): Promise<L1ToL2Message[]>;
 }
 
+/**
+ * A simple in-memory implementation of an L1 to L2 message store
+ * that handles message duplication.
+ */
+export class L1ToL2MessageStore {
```

Comment:
Would not be the place where i expect to find this. Feels wrong here as it is not really a type we are passing around, but a store for "stuff" of this type. Seems closer to the `archiver_store`.